### PR TITLE
Optimized closing characters.

### PR DESCRIPTION
**Pull request by PrestonHager**
When typing a closing character for example the double quote (`"`), any closing characters inside this are also closed. This is good unless the characters are escaped by a backslash (`\`). When typing `"\"` you will get a doubly closed quote, such that the editor displays `"\"""`. This is an invalid string in most editors. Instead, the editor should display `"\""` which is a valid string. This request fixes this by testing for a backslash in the character before the cursor.

### CODE REVIEW

**Comment by antonmedv** [comment]
If you type `"` in the end of `"some str` you still get wrong result. I think this corner case just can be committed. Also what if escape char is different? For example in Bash it is `'''`. And in Joylang it is `^`.

**Comment by PrestonHager** [comment]
Can we add another option for the escape character then? All the large code editors allow for escape characters to be used inside of strings like this. For more efficiency, you can also change the `codeBefore` to just get the first character before the cursor instead of all of the code before. Same with `codeAfter`.

**Comment by PrestonHager** [comment]
Also, CodeJar tends to be used with more common languages like HTML, JS, Python, C, than with Bash or Joylang. In this sense, these are the corner cases that can be ignored as most common languages use the backslash to escape characters.

**Comment by antonmedv** [comment]
Can you upload your solution to https://codesandbox.io/? I'd like to test it.

**Comment by PrestonHager** [comment]
Yeah, here it is: [https://k78pn.csb.app/](https://k78pn.csb.app/)

**Comment by antonmedv** [comment]
Yes, works really good. Just wondering is it make sense to also escape brackes? 
```
(\()
```
Maybe apply this only to `'` and `"`? What do you think?

**Comment by PrestonHager** [comment]
Yeah, that seems more ergonomic, it does add one extra condition but that shouldn't harm processing time too much, especially in more modern browsers. I added that condition to the fork by adding an escapeCharacter condition.

File: codejar.ts [hunk 1]
```diff
@@ -249,13 +249,14 @@ export function CodeJar(editor: HTMLElement, highlight: (e: HTMLElement) => void
     const open = `([{'"`
     const close = `)]}'"`
     const codeAfter = afterCursor()
-    const codeBefore = beforeCursor();
-    if (close.includes(event.key) && codeBefore.substr(codeBefore.length-1) !== "\\" && codeAfter.substr(0, 1) === event.key) {
+    const codeBefore = beforeCursor()
+    var escapeCharacter = `'"`.includes(event.key) && codeBefore.substr(codeBefore.length-1) !== "\\"
```
Comment (antonmedv) [inline codejar.ts:253]: Use const.
